forge-game/src/main/java/forge/game/ability/effects/CloneEffect.java

```diff
@@ -199,6 +199,8 @@ public void run() {
             if (sa.hasParam("RememberCloneOrigin")) {
                 tgtCard.addRemembered(cardToCopy);
             }
+            // spire
+            tgtCard.setChosenColorID(cardToCopy.getChosenColorID());
```

Comment:
@Hanmac
wouldn't it be better if the field was in CardState?
then it doesn't need to be copied around manually (which could also break source since it keeps old values overwritten even if clone effect ends)

Reply:
@tool4ever  @Hanmac  is there anything left needed except for the change if this will be on the cardstate or not? It's working on the player side now (except the AI part since it doesn't generete it's deck on the fly and put the code there what's the best mana combo to use according to deck land stats to be compliant on rules if that's really needed),

Reply:
I don't know about anything else right now

Reply:
Ok I'll just merge this and I'll just update it if needed more changes.